nuxeo-core/nuxeo-core-mongodb/src/test/java/org/nuxeo/ecm/core/mongodb/configuration/TestMongoDBBackEnd.java

```diff
@@ -0,0 +1,161 @@
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.mongodb.MongoDBConnectionService;
+import org.nuxeo.runtime.mongodb.MongoDBFeature;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+import com.google.inject.Inject;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+
+/**
+ * Work In Progress: These tests should be separated to Configuration Test (shouldCreateUniqueIndex) and a new once to
+ * check the MongoDBBackend like TestSQLBackend. Or We can create a new class test to make the check on SQL (postgresql)
+ * and DBS (mongo) to discuss with the team. org.nuxeo.ecm.core.storage.mongodb can hold the back end test, but we have
+ * a cyclic maven dependency module nuxeo-core-test depends on org.nuxeo.ecm.core.storage.mongodb to build
+ * {@link org.nuxeo.ecm.core.test.StorageConfiguration} and this once need {@link CoreFeature}.
+ * <p>
+ * {@link CoreFeature} configure / load {@link org.nuxeo.ecm.core.test.StorageConfiguration} we should find a way to
+ * deploy others custom contribution {@link org.nuxeo.ecm.core.storage.mongodb.MongoDBRepositoryService} to easily
+ * enable / disable the contraints name checking in Tests .... perhaps we need to implement merge on the contribution
+ * 
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ MongoDBFeature.class, CoreFeature.class })
+@Deploy("org.nuxeo.ecm.core.mongodb.test:OSGI-INF/mongodb-disable-renaming-listener-test.xml")
+public class TestMongoDBBackEnd {
+
+    protected static final String DOCUMENT_NAME = "anyFile";
+
+    @Inject
+    protected CoreSession coreSession;
+
+    @Test
+    public void shouldCreateUniqueIndex() {
+        MongoDBConnectionService mongoService = Framework.getService(MongoDBConnectionService.class);
+        MongoDatabase database = mongoService.getDatabase(REPOSITORY_CONNECTION_PREFIX + "test");
+        MongoCollection<Document> collection = database.getCollection("test");
+        Optional<Document> index = StreamSupport.stream(collection.listIndexes().spliterator(), false)
+                                                .filter((d) -> d.get("key", Document.class).containsKey(KEY_PARENT_ID)
+                                                        && d.get("key", Document.class).containsKey(KEY_NAME))
+                                                .findAny();
+
+        assertTrue(index.isPresent());
+        Document document = index.get();
+        assertNotNull(document);
+        assertTrue((Boolean) document.get("unique"));
+    }
+
+    @Test
+    public void shouldFailsWhenCreateExistingChildNameDocument() {
+        DocumentModel domain = coreSession.createDocumentModel("/", "Folder", "Folder");
+        domain = coreSession.createDocument(domain);
+        addPermissions(domain, "james");
+        addPermissions(domain, "bob");
```

Comment:
In what way is the permission stuff relevant to the test, which is about uniqueness of parentid+childname? I'd remove all the ACL stuff and the use of various sessions for different users.

Reply:
i added because i had an exception `Privilege 'AddChildren' is not granted to 'bob'` when i tried to create the document [here](https://github.com/nuxeo/nuxeo/pull/3632/files/c91239adfefe45b863c6eb6b4f90bebf2b914289#diff-4f5ec1ee49aed6c5699ddd058d91fa6eR107)